File: iOS/KinKitchen/KinKitchen/Componets/Cards/KinImageCard.swift
//
//  KinImageCard.swift
//  KinKitchen
//
//  Created by Greg Hudler on 8/26/26.
//

import SwiftUI

struct KinImageCard: View {
    let image: Image
    let title: String
    let subtitle: String?

    var body: some View {
        VStack(alignment: .leading, spacing: KinSpacing.medium) {
            image
                .resizable()
                .scaledToFill()
                .frame(height: 160)
                .clipped()
                .clipShape(
                    RoundedRectangle(cornerRadius: KinRadius.medium)
                )

            Text(title)
                .font(KinTypography.title3)
                .foregroundStyle(KinColors.primaryText)

            if let subtitle {
                Text(subtitle)
                    .font(KinTypography.subheadline)
                    .foregroundStyle(KinColors.secondaryText)
            }
        }
        .padding(KinSpacing.large)
        .background(KinColors.surface)
        .clipShape(
            RoundedRectangle(cornerRadius: KinRadius.large)
        )
    }
}

File: Android/app/src/main/java/com/pushtomaindev/kinkitchen/components/cards/KinImageCard.kt
package com.pushtomaindev.kinkitchen.components.cards

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.painter.Painter
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.unit.dp
import com.pushtomaindev.kinkitchen.ui.theme.*

/** Mirrors iOS `KinImageCard`. */
@Composable
fun KinImageCard(
    painter: Painter,
    title: String,
    modifier: Modifier = Modifier,
    subtitle: String? = null,
    contentDescription: String? = null,
) {
    Column(
        verticalArrangement = Arrangement.spacedBy(KinSpacing.medium),
        modifier = modifier
            .clip(KinRadius.largeShape)
            .background(KinColors.surface)
            .padding(KinSpacing.large),
    ) {
        Image(
            painter = painter,
            contentDescription = contentDescription,
            contentScale = ContentScale.Crop,
            modifier = Modifier
                .fillMaxWidth()
                .height(160.dp)
                .clip(KinRadius.mediumShape),
        )

        Text(title, style = KinTypography.title3, color = KinColors.primaryText)

        if (subtitle != null) {
            Text(subtitle, style = KinTypography.subheadline, color = KinColors.secondaryText)
        }
    }
}
